methods.py

```diff
@@ -1048,6 +1048,51 @@ def progress_finish(target, source, env):
     AlwaysBuild(progress_finish_command)
 
 
+def validate_target_filename(suffix):  # Check if target file can be written to.
+    maximum_attempts = 10
+    if os.access("bin/godot" + suffix, os.F_OK) and not os.access("bin/godot" + suffix, os.W_OK):
+        suffixes = suffix.split(".")  # Split suffix string so we can add suffix number later
+        suffix_last = suffixes.pop()  # Store last suffix, should be the file extension.
+
+        suffix_number = 1  # Suffix number. Will increase until we find a working path to write to
+        access_ok = False  # Is the new suffix ok to write to?
+
+        old_suffix = suffix  # Store old suffix to use in print statement
+
+        while not access_ok:
+            new_suffix = ""
+
+            for suffix_part in suffixes:
+                new_suffix += suffix_part + "."
+            # After this for loop, the new_suffix will have everything except the file extension
+            new_suffix += str(suffix_number) + "."  # Add suffix number
+            new_suffix += suffix_last  # Add file extension
+
+            if not os.access("bin/godot" + new_suffix, os.F_OK) or os.access(
+                "bin/godot" + new_suffix, os.W_OK
+            ):  # Check if this path can be written to
+                access_ok = True
+                # Success!
+                suffix = new_suffix
+                break
+            suffix_number += 1
+            if suffix_number > maximum_attempts:
+                print(
+                    "'bin/godot"
+                    + old_suffix
+                    + "' cannot be written to at this time. This can be caused when trying to compile to an executable that is currently open. Close Godot and try again."
+                )
```

Comment:
```suggestion
    if os.access("bin/godot" + suffix, os.F_OK) and not os.access("bin/godot" + suffix, os.W_OK):
        suffixes = suffix.split(".")  # Split suffix string so we can add suffix number later
        extension = suffixes.pop()  # Store last suffix, should be the file extension.

        old_suffix = suffix  # Store old suffix to use in print statement
        suffix = ""

        MAXIMUM_ATTEMPTS = 10
        for suffix_number in range(1, MAXIMUM_ATTEMPTS + 1):
            new_suffix = ".".join(suffixes) + "." + str(suffix_number) + "." + extension
            if not os.access("bin/godot" + new_suffix, os.F_OK) or os.access(
                "bin/godot" + new_suffix, os.W_OK
            ):  # Check if this path can be written to
                suffix = new_suffix
                break

        if not suffix:
            print(
                "'bin/godot"
                + old_suffix
                + "' cannot be written to at this time. This can be caused when trying to compile to an executable that is currently open. Close Godot and try again."
            )
        else:
            print(
                "'bin/godot"
                + old_suffix
                + "' cannot be written to at this time. Compiling to 'bin/godot"
                + suffix
                + "' instead."
            )
    return suffix
```

Reply:
Thanks!